packages/expo-navigation-bar/android/src/main/java/expo/modules/navigationbar/NavigationBarModule.kt

```diff
@@ -1,201 +1,167 @@
 package expo.modules.navigationbar
 
 import android.app.Activity
-import android.content.Context
 import android.graphics.Color
 import android.os.Build
 import android.os.Bundle
 import android.view.View
 import android.view.WindowInsets
 import androidx.core.view.ViewCompat
 import androidx.core.view.WindowInsetsControllerCompat
-import expo.modules.core.ExportedModule
-import expo.modules.core.ModuleRegistry
-import expo.modules.core.Promise
-import expo.modules.core.errors.CurrentActivityNotFoundException
-import expo.modules.core.interfaces.ActivityProvider
-import expo.modules.core.interfaces.ExpoMethod
-import expo.modules.core.interfaces.services.EventEmitter
+import expo.modules.kotlin.Promise
+import expo.modules.kotlin.exception.Exceptions
+import expo.modules.kotlin.modules.Module
+import expo.modules.kotlin.modules.ModuleDefinition
+import expo.modules.navigationbar.singletons.NavigationBar
 
-class NavigationBarModule(context: Context) : ExportedModule(context) {
+class NavigationBarModule : Module() {
+  override fun definition() = ModuleDefinition {
+    Name("ExpoNavigationBar")
 
-  private lateinit var mActivityProvider: ActivityProvider
-  private lateinit var mEventEmitter: EventEmitter
+    Events(VISIBILITY_EVENT_NAME)
 
-  override fun getName(): String {
-    return NAME
-  }
-
-  override fun onCreate(moduleRegistry: ModuleRegistry) {
-    mActivityProvider = moduleRegistry.getModule(ActivityProvider::class.java)
-      ?: throw IllegalStateException("Could not find implementation for ActivityProvider.")
-    mEventEmitter = moduleRegistry.getModule(EventEmitter::class.java) ?: throw IllegalStateException("Could not find implementation for EventEmitter.")
-  }
-
-  // Ensure that rejections are passed up to JS rather than terminating the native client.
-  private fun safeRunOnUiThread(promise: Promise, block: (activity: Activity) -> Unit) {
-    val activity = mActivityProvider.currentActivity
-    if (activity == null) {
-      promise.reject(CurrentActivityNotFoundException())
-      return
-    }
-    activity.runOnUiThread {
-      block(activity)
+    AsyncFunction("setBackgroundColorAsync") { color: Int, promise: Promise ->
+      safeRunOnUiThread {
+        NavigationBar.setBackgroundColor(it, color, { promise.resolve(null) }, { m -> promise.reject(NavigationBarException(m)) })
+      }
     }
```

Comment:
```suggestion
    AsyncFunction("setBackgroundColorAsync") { color: Int, promise: Promise ->
      NavigationBar.setBackgroundColor(it, color, { promise.resolve(null) }, { m -> promise.reject(NavigationBarException(m)) })
    }.runOnQueue(Queues.MAIN)
```
You can use `runOnQueue` here and other places like this ;)

Reply:
🤦‍♂️😅